# Publish all joint states -----------------------------------
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

arm_controller:
  type: position_controllers/JointTrajectoryController
  joints:
    - joint1
    - joint2
    - joint5
  constraints:
    stopped_velocity_tolerance: 0.01
  stop_trajectory_duration: 0.5
  state_publish_rate:  125
  action_monitor_rate: 10

# Note: You MUST load these PID parameters for all joints that are using the
# PositionJointInterface, otherwise the arm + gripper will act like a giant
# parachute, counteracting gravity, and causing some of the wheels to lose
# contact with the ground, so the robot won't be able to properly navigate. See
# https://github.com/ros-simulation/gazebo_ros_pkgs/issues/612
gazebo_ros_control:
  pid_gains:
    # these gains are used by the gazebo_ros_control plugin
    joint1:
      p: 20.0
      i: 0.1
      d: 0.0
      i_clamp: 0.2
      antiwindup: false
      publish_state: true
    joint2:
      p: 20.0
      i: 0.1
      d: 0.0
      i_clamp: 0.2
      antiwindup: false
      publish_state: true
    joint5:
      p: 100.0
      i: 0.01
      d: 10.0
      i_clamp: 0.2
      antiwindup: true
      publish_state: true
    # the following gains are used by the gazebo_mimic_joint plugin
    joint3:
      p: 20.0
      i: 0.1
      d: 0.0
      i_clamp: 0.2
      antiwindup: false
      publish_state: true
    joint4:
      p: 20.0
      i: 0.1
      d: 0.0
      i_clamp: 0.2
      antiwindup: false
      publish_state: true
    joint6:
      p: 100.0
      i: 0.01
      d: 10.0
      i_clamp: 0.2
      antiwindup: true
      publish_state: true
